### PR TITLE
core workflow: consolidate search results info bar actions

### LINKED ISSUE #38438: Consolidate action/create/extension buttons into one button
Put actions related to search results in the existing "Create" button, and either rename it or give it an "action" icon type.

Remaining offending actions:
- Extensions buttons (ex: export to CSV)
- Save search
- Any extensions that create buttons in this area when activated

Per @paulo we'll create a generic three dot action button and segment out the action types as in the avatar dropdown.

See the three dot button and opened dropdown below ([Figma](https://www.figma.com/file/8MkIcpDtjv6tqgxZFcEQDt/Core-workflow-punch-list?node-id=27%3A7655)):

image

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #38438 

FYI: I decided to duplicate some of the code for easier future removal. I didn't want to leave a big footprint by trying to adjust existing components.

## Demo

https://user-images.githubusercontent.com/6417322/178445655-0aae6a3c-d529-488c-80cb-a1750324d993.mov

## Test plan

* Do a search
* There should be a single menu in the top right corner of the search results, containing:
  * Collapse/Expand search result matches
  * Create monitor
  * Create context
  * Create insight
  * Create batch change
  * Save search

## App preview:

- [Web](https://sg-web-rn-consolidate-search-results.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jykgpbfjee.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/search/results/SearchActionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                        className={classNames('d-flex align-items-center text-decoration-none')}
+                        aria-label={`${isExpanded ? 'Close' : 'Open'} search actions menu`}
+                        variant="secondary"
+                        outline={true}
+                        size="sm"
+                    >
+                        <Icon aria-hidden={true} svgPath={mdiDotsHorizontal} />
+                    </MenuButton>
+                </li>
+                <MenuList tabIndex={0} position={Position.bottomEnd} aria-label="Search Actions. Open menu">
```
Comment (limitedmage) [inline client/web/src/search/results/SearchActionsMenu.tsx:49]: Should the menu list go inside the li (sibling to the menu button)?

File: client/web/src/search/results/SearchActionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                            data-tooltip={
+                                authenticatedUser && !canCreateMonitor
+                                    ? 'Code monitors only support type:diff or type:commit searches.'
+                                    : undefined
+                            }
+                            aria-label={
+                                authenticatedUser && !canCreateMonitor
+                                    ? 'Code monitors only support type:diff or type:commit searches.'
+                                    : undefined
+                            }
```
Comment (limitedmage) [inline client/web/src/search/results/SearchActionsMenu.tsx:89]: This `aria-label` doesn't seem right to me. I think removing it is better. I think that will have the screen reader read the text content, disabled state, and tooltip.

File: client/web/src/search/results/SearchActionsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                                    ? { svgPath: createAction.icon }
+                                    : { as: createAction.icon })}
+                            />
+                            {createAction.label}
+                        </MenuLink>
+                    ))}
+                    {createCodeMonitorAction && (
+                        <MenuLink
+                            as={Link}
+                            disabled={!authenticatedUser || !canCreateMonitor}
```
Comment (limitedmage) [inline client/web/src/search/results/SearchActionsMenu.tsx:79]: Should we show the menu items as disabled if the user is not signed in, or remove it entirely?

**Review by limitedmage — APPROVED** [review]
Looks great! It's a shame the work to add extension actions here is too complex for this first iteration. Certain users will see this instead of the clean look:

image

File: client/web/src/search/results/SearchActionsMenu.tsx (hunk 3 above)
Comment (novoselrok) [reply client/web/src/search/results/SearchActionsMenu.tsx:79]: If the user is logged out: we remove context, batch changes, and insight buttons and disable the monitor and save searches buttons.

**Pull request merged**